could not baseline diagnostics on Python 3.14 docker container

### Description

This seems to be exactly the same as #1322. It works fine locally but in CI I am getting a warning with a non-zero exit code:

could not baseline diagnostics for the following files because they are located outside the project root directory (/src):
  - /app/lib/python3.14/site-packages/basedpyright/dist/typeshed-fallback/stdlib/string/templatelib.pyi
  0 errors, 0 warnings, 0 notes

This could be 100% irrelevant but Claude Code thinks it has something to do with a conflict with string.templatelib added in Python 3.14 🤷

I'm facing the same issue, I will try to share any information that can help.

I only face the issue when running on the CI or inside of a container, I believe the difference is the fact that locally I have the .venv in my project dir.

The logs from my CI runner ⏬ 
```
basedpyright --stats
Loading pyproject.toml file at /builds/myproject/pyproject.toml
Assuming Python version 3.14.3.final.0
Found 62 source files
could not baseline diagnostics for the following files because they are located outside the project root directory (/builds/myproject):
- /root/.cache/pypoetry/virtualenvs/myproject-ld5ZfdwO-py3.14/lib/python3.14/site-packages/basedpyright/dist/typeshed-fallback/stdlib/string/templatelib.pyi
basedpyright 1.38.4
based on pyright 1.1.408
0 errors, 0 warnings, 0 notes
Completed in 2.894sec
Analysis stats
Total files parsed and bound: 532
Total files checked: 62
Timing stats
Find Source Files:    0.01sec
Read Source Files:    0.02sec
Tokenize:             0.28sec
Parse:                0.37sec
Resolve Imports:      0.16sec
Bind:                 0.54sec
Check:                1.3sec
Detect Cycles:        0sec
make: *** [Makefile:55: lint-basedpyright] Error 3
```

* Python version: 3.14.3
* Basepyright version: 1.38.4
* Command: `basedpyright --stats`
* PyProject.toml (config below)

```
[tool.pyright]
typeCheckingMode = "standard"
include = ["src", "tests"]
exclude = ["**/.venv", "**/__pycache__", "**/node_modules"]
```

--- a/.vscode/launch.json
+++ b/.vscode/launch.json
@@ -36,11 +36,15 @@
         {
             "name": "Pyright current file",
             "type": "node",
+            "runtimeExecutable": "${workspaceFolder}/.venv/bin/node",
             "request": "launch",
             "preLaunchTask": "npm: build:cli:dev",
             "console": "integratedTerminal",
             "program": "${workspaceFolder}/packages/pyright/index.js",
-            "args": ["${file}"]
+            "args": ["${file}"],
+            "windows": {
+                "runtimeExecutable": "${workspaceFolder}/.venv/Scripts/node.exe"
+            }
         },
         {
             "name": "Pyright CLI (pyright-internal)",

--- a/packages/pyright-internal/src/analyzer/program.ts
+++ b/packages/pyright-internal/src/analyzer/program.ts
@@ -1015,7 +1015,13 @@ export class Program {
         const fileDiagnostics: FileDiagnostics[] = this._removeUnneededFiles();
 
         this._sourceFileList.forEach((sourceFileInfo) => {
-            if (this._shouldCheckFile(sourceFileInfo)) {
+            if (
+                this._shouldCheckFile(sourceFileInfo) &&
+                // we never want to show diagnostics in non-user files. ideally we would check this as part of
+                // `_shouldCheckFile` but there are other places where that function is used that seem to require
+                // non-user files to be checked otherwise they won't show up in completions
+                isUserCode(sourceFileInfo)
+            ) {
                 let diagnostics = sourceFileInfo.sourceFile.getDiagnostics(
                     options,
                     reportDeltasOnly ? sourceFileInfo.diagnosticsVersion : undefined

--- a/packages/pyright-internal/src/tests/sourceFile.test.ts
+++ b/packages/pyright-internal/src/tests/sourceFile.test.ts
@@ -18,6 +18,7 @@ import { createServiceProvider } from '../common/serviceProviderExtensions';
 import { Uri } from '../common/uri/uri';
 import { parseAndGetTestState } from './harness/fourslash/testState';
 import { BaselineHandler } from '../baseline';
+import { tExpect } from 'typed-jest-expect';
 
 test('Empty', () => {
     const filePath = combinePaths(process.cwd(), 'tests/samples/test_file1.py');
@@ -96,7 +97,7 @@ test('Empty Open file', () => {
     assert.strictEqual(state.workspace.service.test_program.getSourceFile(marker.fileUri)?.getFileContent(), '');
 });
 
-test('No unexpected user files', () => {
+describe('no builtin libraries should be treated as user code', () => {
     const code = `
 // @filename: pyrightconfig.json
 //// {
@@ -111,7 +112,19 @@ test('No unexpected user files', () => {
     const marker = state.getMarkerByName('marker');
     while (state.workspace.service.test_program.analyze());
 
-    const userFiles = state.workspace.service.test_program.getUserFiles();
-    assert.strictEqual(userFiles.length, 1);
-    assert.strictEqual(userFiles[0].uri.toString(), marker.fileUri.toString());
+    test('No unexpected user files', () => {
+        const userFiles = state.workspace.service.test_program.getUserFiles();
+        assert.strictEqual(userFiles.length, 1);
+        assert.strictEqual(userFiles[0].uri.toString(), marker.fileUri.toString());
+    });
+
+    test('No unexpected files with diagnostics', () => {
+        const tempFile = new RealTempFile();
+        const fs = createFromRealFileSystem(tempFile);
+        const serviceProvider = createServiceProvider(tempFile, fs);
+        const configOptions = new ConfigOptions(Uri.file(process.cwd(), serviceProvider));
+        const fileDiagnostics = state.workspace.service.test_program.getDiagnostics(configOptions);
+        tExpect(fileDiagnostics.length).toBe(1);
+        tExpect(fileDiagnostics[0].fileUri.toString()).toBe(marker.fileUri.toString());
+    });
 });

